### ISSUE TITLE
Error using SHA-1?

### ISSUE THREAD

**Issue by cjvanlissa** [r-lib/gert#172]
I'm using a clean install of RStudio, R, and Git on a Windows 10 system, with an SSH key for GitHub set up in the RStudio settings. Still, `gert::push()` results in the following error:

```
Error in libgit2::git_remote_push : 
  ERROR: You're using an RSA key with SHA-1, which is no longer allowed. Please use a newer client or a different key type.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
```

**Comment by jeroen** [comment]
Hmm interesting, I have not seen this before. How exactly did you generate/setup your ssh key? We may have to switch to another key type for newly generated keys.

**Comment by cjvanlissa** [comment]
Using the button "Create RSA key" in RStudio.. :/

**Comment by cjvanlissa** [comment]
Surprisingly, I can `git push remote origin` from the command line..

**Comment by jeroen** [comment]
I opened an issue here for studio: https://github.com/rstudio/rstudio/issues/10807

**Comment by jeroen** [comment]
@cjvanlissa i've updated the version of libgit2 in the latest version of gert. I honestly don't think that will fix it, but do you mind trying nevertheless? The easiest way to install new gert is:

```r
install.packages("gert", repos = "https://ropensci.r-universe.dev")
```

**Comment by GuilhermeFCO** [comment]
Hi @jeroen, estou tentando utilizar gert no Windows 10 mas estou recebendo este erro quando tento fazer clone utilizando SSH. On linux it works fine, even with the key type ed25519.

```
Unable to find ssh-agent (SSH_AUTH_SOCK undefined)
Trying to authenticate 'git' using provided ssh-key...
Failed to authenticate over SSH. You either need to provide a key or setup ssh-agent
Error in libgit2::git_clone : Authentication failure
````

**Comment by sda030** [comment]
> I'm using a clean install of RStudio, R, and Git on a Windows 10 system, with an SSH key for GitHub set up in the RStudio settings. Still, `gert::push()` results in the following error:
> 
> ```
> Error in libgit2::git_remote_push : 
>   ERROR: You're using an RSA key with SHA-1, which is no longer allowed. Please use a newer client or a different key type.
> Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
> ```

I had a similar issue. Tried many things. Turned out that I had to rename my id_rsa and id_rsa.pub into id_ecdsa and id_ecdsa.pub (inspired by rule 3 on https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys).

**Comment by jeroen** [comment]
This error should be fixed in gert 1.8.0 (see the NEWS file)
